rpc/block_svc_routines.c

```diff
@@ -3900,8 +3899,12 @@ block_create_cli_1_svc_st(blockCreateCli *blk, struct svc_req *rqstp)
   }
 
   if (glusterBlockCreateEntry(glfs, blk, gbid, &errCode, &errMsg)) {
-    LOG("mgmt", GB_LOG_ERROR, "%s volume: %s host: %s",
-        FAILED_CREATING_FILE, blk->volume, blk->block_hosts);
+    LOG("mgmt", GB_LOG_ERROR, "%s volume: %s block: %s file: %s host: %s",
+        FAILED_CREATING_FILE, blk->volume, blk->block_name, gbid, blk->block_hosts);
+    if (glusterBlockDeleteMetaFile(glfs, blk->volume, blk->block_name)) {
```

Comment:
Even metafile deletion is happening in the  glusterBlockCreateEntry() function? Could you also move the LOG inside that function? May be in another patch it is better to remove glfs_chdir() usage from glfs-operations.c, that way we don't have to worry about pwd ever.